File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        """
        :type target: int
        :type nums: List[int]
        :rtype: int
        """
        min_len=float('inf')
        sum=0
        k=0
        for i in range(len(nums)):
            sum+=nums[i]
            while sum>=target:
                min_len=min(min_len,i-k+1)
                sum-=nums[k]
                k+=1
            
        return 0 if(min_len==float('inf')) else min_len

File: 209-minimum-size-subarray-sum/minimum-size-subarray-sum.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int sum=0,len=Integer.MAX_VALUE;
        for(int i=0,k=0;i<nums.length;i++){
            sum+=nums[i];
            while(sum>=target){
                len=Math.min(i-k+1,len);
                sum-=nums[k++];
            }
            // if(sum==target){
            //     len=Math.min(i-k+1,len);

            // }
        }
        
        return (len==Integer.MAX_VALUE)?0:len;
    }
}